Testing: Unblock `new_*` testing on MPS

malfet opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #136987
* #137004
* __->__ #137003
* #136986
* #136985
* #136984
* #136983
* #136982
* #136981

By changing `other_dtype` to `torch.half` rather than `double` in
`sample_inputs_new_fns` if MPS is available

malfet committed:
Testing: Unblock `new_*` testing on MPS

By changing `other_dtype` to `torch.half` rather than `double` in
`sample_inputs_new_fns` if MPS is available

[ghstack-poisoned]

Skylion007 commented on torch/testing/_internal/common_methods_invocations.py:
```diff
@@ -1791,6 +1791,7 @@ def error_inputs_margin_ranking_loss(op, device, **kwargs):
                      error_regex='margin_ranking_loss : All input tensors should')
 
 def sample_inputs_new_fns(self, device, dtype, requires_grad, *, is_strided=False, **kwargs):
+    other_dtype = torch.half if torch.backends.mps.is_available() else torch.double
```
is there a better variable we can use to probe for float64 support?

malfet replied:
No, not really. Or at least I can't think of any. Also, this test perhaps should be further extended to pick different dtype if input dtype is torch.double/`torch.half`

malfet committed:
Update on "Testing: Unblock `new_*` testing on MPS"


By changing `other_dtype` to `torch.half` rather than `double` in
`sample_inputs_new_fns` if MPS is available

[ghstack-poisoned]
